iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/extractor/dataregion/realtime/listener/PipeInsertionDataNodeListener.java

```diff
@@ -141,9 +145,14 @@ public void listenToHeartbeat(boolean shouldPrintMessage) {
                 PipeRealtimeEventFactory.createRealtimeEvent(key, shouldPrintMessage)));
   }
 
-  public void listenToDeleteData(DeleteDataNode node) {
-    dataRegionId2Assigner.forEach(
-        (key, value) -> value.publishToAssign(PipeRealtimeEventFactory.createRealtimeEvent(node)));
+  public DeletionResource listenToDeleteData(DeleteDataNode node, String regionId) {
+    PipeRealtimeEvent realtimeEvent = PipeRealtimeEventFactory.createRealtimeEvent(node, regionId);
+    dataRegionId2Assigner.forEach((key, value) -> value.publishToAssign(realtimeEvent));
```

Comment:
In fact, we better tactle the deletion just like the insertion... Previously I just let all the assigners to publish it, then a "delete" operation will be captured "dataRegionNum" times and each time send to all the assigner.. If there are 100 regions, this may be send to the receiver 10000 times.. Can correct my bug by the way :-(

Reply:
good catch，fixed